specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -251,6 +257,51 @@ function Merging({
   );
 }
 
+function MergeButton<SCHEMA extends AnySchema>({
+  id,
+  mergeResource,
+}: {
+  readonly id: (suffix: string) => string;
+  readonly mergeResource: SpecifyResource<SCHEMA>;
+}): JSX.Element {
+  const [formId] = React.useState(id('form'));
+  const [saveBlocked, setSaveBlocked] = React.useState(false);
+  const [showSaveBlockedDialog, setShowBlockedDialog] = React.useState(false);
+
+  React.useEffect(() => {
+    setSaveBlocked(false);
+    return resourceOn(
+      mergeResource,
+      'blockersChanged',
+      (): void => {
+        const onlyDeferredBlockers = Array.from(
+          mergeResource.saveBlockers?.blockingResources ?? []
+        ).every((resource) => resource.saveBlockers?.hasOnlyDeferredBlockers());
+        setSaveBlocked(!onlyDeferredBlockers);
+      },
+      true
+    );
+  }, [mergeResource]);
+
+  return (
+    <>
+      {!saveBlocked ? (
+        <Submit.Blue form={formId}>{treeText.merge()}</Submit.Blue>
+      ) : (
+        <Submit.Red className="cursor-not-allowed">
+          {treeText.merge()}
+        </Submit.Red>
+      )}
+      {showSaveBlockedDialog && (
```

Comment:
looks like a bug. you call `setShowBlockedDialog(false)`, but you never call `setShowBlockedDialog(true)` thus this dialog will never be displayed